tikv-client/src/main/java/com/pingcap/tikv/Snapshot.java

```diff
@@ -106,7 +85,7 @@ public ByteString get(ByteString key) {
    * @return Row iterator to iterate over resulting rows
    */
   public Iterator<Row> tableRead(TiDAGRequest dagRequest, List<RegionTask> task) {
-    if (dagRequest.isIndexScan()) {
+    if (dagRequest.isDoubleRead()) {
```

Comment:
the condition is changed into `isDoubleRead` because we need `getHandleIterator` only when index double read.